### ISSUE TITLE
🍅 짭짤이 논문 모아보기 (XAI)

### ISSUE THREAD

**Issue by long8v** [long8v/PTIR#167]

**Comment by long8v** [comment]
## Network Dissection
Quantifying Interpretability of Deep Visual Representations

https://arxiv.org/pdf/1704.05796.pdf

unit이 어떻게 이렇게 label이 달렸나가 궁금했는데 선행연구(https://arxiv.org/pdf/1412.6856.pdf)에서 part에 대한 annotation을 단 거 였음..

<img width="921" alt="image" src="https://github.com/long8v/PTIR/assets/46675408/9b45bc4f-095d-4e2a-9d36-f939199090ac">

**Comment by long8v** [comment]
## ALBEF 

<img width="939" alt="image" src="https://github.com/long8v/PTIR/assets/46675408/37fa7e94-d596-489b-abe3-e99015dd8357">

<img width="922" alt="image" src="https://github.com/long8v/PTIR/assets/46675408/b8b59dae-2051-4bb4-aaec-e97fb28220ad">

<img width="772" alt="image" src="https://github.com/long8v/PTIR/assets/46675408/d9452026-69c8-43f5-9450-62bd39968841">

**Comment by long8v** [comment]
## Label Words are Anchors: An Information Flow Perspective for Understanding In-Context Learning

In-Context Learning 에서 shallow layer에서 Label이 관련된 정보를 aggregate하고 LLM의 Final prediction에서 다시 그 Label들을 참고해서 예측한다는 관찰
<img width="477" alt="image" src="https://github.com/long8v/PTIR/assets/46675408/767a7bb8-2531-4119-823f-6b50f8805b58">

**Comment by long8v** [comment]
## Contrastive Explanations for Model Interpretability
![image](https://github.com/long8v/PTIR/assets/46675408/671673e4-d13a-4add-9223-e86a0070d5b2)

```python
import torch
import torch.nn as nn
import torch.nn.functional as F

class ContrastiveExplanation(nn.Module):
    def __init__(self, model):
        super(ContrastiveExplanation, self).__init__()
        self.model = model
    
    def forward(self, input_ids, attention_mask, fact_class, foil_class):
        # 입력 표현 생성
        hiddens = self.model.encoder(input_ids, attention_mask)[0]
        
        # 최종 예측을 위한 선형 변환
        logits = self.model.classifier(hiddens[:, 0, :])
        
        # Fact와 Foil 클래스에 해당하는 가중치 추출
        W_fact = self.model.classifier.weight[fact_class]
        W_foil = self.model.classifier.weight[foil_class]
        
        # 대조 방향 계산
        u = W_fact - W_foil
        
        # 대조 공간으로 투영
        P_u = u.unsqueeze(0) @ u.unsqueeze(1) / torch.norm(u)**2
        h_contrast = hiddens @ P_u.squeeze(1)
        
        # 토큰별 중요도 계산
        token_importance = torch.sum(hiddens * h_contrast, dim=-1)
        token_importance = token_importance.masked_fill(attention_mask == 0, float('-inf'))
        token_importance = F.softmax(token_importance, dim=-1)
        
        return token_importance

# 설명 생성 예시
model = ContrastiveExplanation(pretrained_model)
input_ids = torch.tensor([[101, 2023, 2003, 1037, 3231, 102]])
attention_mask = torch.tensor([[1, 1, 1, 1, 1, 1]])
fact_class = torch.tensor(0)
foil_class = torch.tensor(1)

token_importance = model(input_ids, attention_mask, fact_class, foil_class)
print(token_importance)
```

**Comment by long8v** [comment]
## VIEScore: Towards Explainable Metrics for Conditional Image Synthesis Evaluation
이미지와 생성된 이미지가 있을 때 score와 함께 설명까지 내보낼 수 있는 task를 제안
<img width="394" alt="image" src="https://github.com/long8v/PTIR/assets/46675408/fa1133f5-76c0-47ec-a643-5f49c201328b">

<img width="885" alt="image" src="https://github.com/long8v/PTIR/assets/46675408/cff64b04-a330-4484-ade0-be67f4ee1ae1">

생성 모델에 특화된 느낌. 
<img width="442" alt="image" src="https://github.com/long8v/PTIR/assets/46675408/b44987f1-4f54-4201-a627-10115f78cca6">

perceptual quality와 semantic consistency로 나누어서 측정.
나로 따지면 semantic consistency만 관심 있음.

<img width="427" alt="image" src="https://github.com/long8v/PTIR/assets/46675408/772e9913-20d6-44e5-bc89-d0305788ff40">

CLIPScore가 GPT4-V보다 human correlation이 매우 낮았다고 함. 
일단 CLIP이 synthetic을 못본 이유도 있을 것 같고, 여기서 설명하기에는 text 전체를 abstraction으로 봐서 그렇다고 하넹 

<img width="428" alt="image" src="https://github.com/long8v/PTIR/assets/46675408/d6c64ae8-9d82-4c01-bef6-e2f3ca21ab1f">

**Comment by long8v** [comment]
## Attention is not Explanation
RNN에서의 attention score를 보고 해석하는건 한계가 있다. 가령 gradient based와 Attention based는 양상이 다르고, output이 똑같지만 attention map은 분포가 매우 다른 counter attention map 같은걸 만들 수 있음. 
결론 부분만 보면 Recurrent가 아닌 구조에서 leave-one-out (LOO) 방법론은 gradient based 방법론이랑 Correlation이 높다고 함. 
<img width="484" alt="image" src="https://github.com/long8v/PTIR/assets/46675408/c5ba6687-00ee-41d4-9c1f-ad43b5b00597">

RNN이 아닌 다른 FFN에서는 이런 경향이 안일어났다고 함
<img width="486" alt="image" src="https://github.com/long8v/PTIR/assets/46675408/819f1840-91dc-48e0-ba83-c4093dd9d374">

**Comment by long8v** [comment]
## Optimizing Relevance Maps of Vision Transformers Improves Robustness

<img width="902" alt="image" src="https://github.com/long8v/PTIR/assets/46675408/a873d31e-1cdb-40ce-ad13-790f9922afa4">

ViT가 분류할 때 spurious cue에 집중(배경을 보고 분류)하는 경향이 있었고, 이를 해결하기 위해 relevance map을 활용해서 foreground에 집중해서 분류하도록 했더니 ood 성능이 좋아졌다는 내용

**Comment by long8v** [comment]
<img width="427" alt="image" src="https://github.com/long8v/PTIR/assets/46675408/c1035fda-ab26-483d-8a9b-95229b26baec">

<img width="876" alt="image" src="https://github.com/long8v/PTIR/assets/46675408/c838036f-54b4-4bdc-82cf-6b7fd5b953b1">

<img width="407" alt="image" src="https://github.com/long8v/PTIR/assets/46675408/31d146ea-60a4-4556-b9b3-2f7407c3ff63">
